joblib/hashing.py

```diff
@@ -194,7 +194,10 @@ def save(self, obj):
                 # https://github.com/numpy/numpy/issues/4983.  The
                 # workaround is to view the array as bytes before
                 # taking the memoryview
-                obj_bytes_view = obj.view(self.np.uint8)
+                if obj.flags.f_contiguous:
+                    obj_bytes_view = obj.T.view(self.np.uint8).T
+                else:
+                    obj_bytes_view = obj.view(self.np.uint8)
```

Comment:
I would prefer something like this:
```python:
if f_contiguous and not c_contiguous:
   # Use transpose version
else:
   # For all other cases use the default
```

Probably not perfect but I don't know if we should do something special with non contiguous arrays.